acme/challenge.go

```diff
@@ -421,31 +425,51 @@ func wireOIDC01Validate(ctx context.Context, ch *Challenge, db DB, jwk *jose.JSO
 		return WrapErrorISE(err, "error updating challenge")
 	}
 
-	parsedIDToken, err := jose.ParseSigned(oidcPayload.IDToken)
-	if err != nil {
-		return WrapErrorISE(err, "invalid OIDC ID token")
-	}
-	oidcToken := make(map[string]interface{})
-	if err := parsedIDToken.UnsafeClaimsWithoutVerification(&oidcToken); err != nil {
-		return WrapErrorISE(err, "failed parsing OIDC id token")
-	}
-
 	orders, err := db.GetAllOrdersByAccountID(ctx, ch.AccountID)
 	if err != nil {
-		return WrapErrorISE(err, "could not find current order by account id")
+		return WrapErrorISE(err, "could not retrieve current order by account id")
 	}
 	if len(orders) == 0 {
 		return NewErrorISE("there are not enough orders for this account for this custom OIDC challenge")
 	}
 
 	order := orders[len(orders)-1]
-	if err := db.CreateOidcToken(ctx, order, oidcToken); err != nil {
+	if err := db.CreateOidcToken(ctx, order, transformedIDToken); err != nil {
 		return WrapErrorISE(err, "failed storing OIDC id token")
 	}
 
 	return nil
 }
 
+func validateWireOIDCClaims(o *wireprovisioner.OIDCOptions, token *oidc.IDToken, wireID wire.ID) (map[string]any, error) {
+	var m map[string]any
+	if err := token.Claims(&m); err != nil {
+		return nil, fmt.Errorf("failed extracting OIDC ID token claims: %w", err)
+	}
+	transformed, err := o.Transform(m)
+	if err != nil {
+		return nil, fmt.Errorf("failed transforming OIDC ID token: %w", err)
+	}
+
+	name, ok := transformed["name"]
+	if !ok {
+		return nil, fmt.Errorf("transformed OIDC ID token does not contain 'name'")
+	}
+	if wireID.Name != name {
+		return nil, fmt.Errorf("invalid 'name' %q after transformation", name)
+	}
+
+	handle, ok := transformed["handle"]
+	if !ok {
+		return nil, fmt.Errorf("transformed OIDC ID token does not contain 'handle'")
+	}
+	if wireID.Handle != handle {
+		return nil, fmt.Errorf("invalid 'handle' %q after transformation", handle)
+	}
```

Comment:
Check with wire the case sensitivity of these conditions.

Reply:
@beltram? Now, the transformations _can_ make them use the right case, but then you would also have these values available in the template. Do we need to support case-sensitivity here?

Reply:
Nah let's not bother with case-sensitivity. Those claim names will already be canonicalized by the template. And since we tend to adhere to the standards we will have them snake_case (lower case by consequence). So what you have here is fine.